service:
  port: 3000

namespaceOverride: is-stag

adminPassword: admin

# Fix "Readiness probe failed: connection refused" error
readinessProbe:
  initialDelaySeconds: 20

dashboardProviders:
  dashboardproviders.yaml:
    apiVersion: 1
    providers:
      - name: 'prometheus'
        type: file
        disableDeletion: false
        updateIntervalSeconds: 10
        editable: true
        options:
          path: /var/lib/grafana/dashboards/default

dashboards:
  default:
    k6:
      gnetId: 19665
      revision: 3
      datasource: Prometheus

datasources:
  datasources.yaml:
    apiVersion: 1
    datasources:
      - name: Loki
        type: loki
        uid: loki
        url: http://loki:3100
        editable: true
        jsonData:
          derivedFields:
            - name: trace_id
              matcherType: label
              matcherRegex: trace_id
              url: "$${__value.raw}"
              datasourceUid: tempo
              urlDisplayLabel: "Trace: $${__value.raw}"

      - name: Tempo
        type: tempo
        uid: tempo
        url: http://tempo:3200
        editable: true
        jsonData:
          tracesToLogsV2:
            customQuery: true
            datasourceUid: loki
            query: '{$${__tags}} | trace_id = "$${__trace.traceId}"'
            tags:
              - key: "service.name"
                value: "service_name"
          serviceMap:
            datasourceUid: prometheus
          tracesToMetrics:
            datasourceUid: prometheus
          search:
            hide: false
          nodeGraph:
            enabled: true
          lokiSearch:
            datasourceUid: loki

      - name: Prometheus
        type: prometheus
        uid: prometheus
        url: http://prometheus:9090
        editable: true
        jsonData:
          timeInterval: 60s
          exemplarTraceIdDestinations:
            - name: trace_id
              datasourceUid: tempo
              urlDisplayLabel: "Trace: $${__value.raw}"